src/examples/java/com/amazonaws/crypto/examples/FileStreamingExample.java

```diff
@@ -40,55 +43,83 @@
+
+        // 2. Retrieve an encryption key. In this example, we generate a random key.
+        //    In practice, you would get a key from an existing store
+        final SecretKey cryptoKey = retrieveEncryptionKey();
+
+        // 3. Instantiate a RawAesKeyring using the random key
+        final Keyring keyring = StandardKeyrings.rawAes("Example", "RandomKey", cryptoKey);
 
-        // Create a JCE master key provider using the random key and an AES-GCM encryption algorithm
-        JceMasterKey masterKey = JceMasterKey.getInstance(cryptoKey, "Example", "RandomKey", "AES/GCM/NoPadding");
+        // 4. Create an encryption context
+        //
+        //    Most encrypted data should have an associated encryption context
+        //    to protect integrity. This sample uses placeholder values.
+        //
+        //    For more information see:
+        //    blogs.aws.amazon.com/security/post/Tx2LZ6WBJJANTNW/How-to-Protect-the-Integrity-of-Your-Encrypted-Data-by-Using-AWS-Key-Management
+        final Map<String, String> encryptionContext = Collections.singletonMap("Example", "FileStreaming");
 
-        // Instantiate the SDK
-        AwsCrypto crypto = new AwsCrypto();
+        // 5. Instantiate the AwsCryptoConfig input to AwsCrypto with the keyring and encryption context
+        final AwsCrypto.AwsCryptoConfig config = AwsCrypto.AwsCryptoConfig.builder()
+                .keyring(keyring)
+                .encryptionContext(encryptionContext)
+                .build();
 
-        // Create an encryption context to identify this ciphertext
-        Map<String, String> context = Collections.singletonMap("Example", "FileStreaming");
+        // 6. Create the encrypting stream. Because the file might be too large to load into memory,
+        //    we stream the data, instead of loading it all at once.
+        final AwsCryptoInputStream encryptingStream =
+                crypto.createEncryptingStream(config, new FileInputStream(srcFile));
 
-        // Because the file might be to large to load into memory, we stream the data, instead of 
-        //loading it all at once.
-        FileInputStream in = new FileInputStream(srcFile);
-        CryptoInputStream<JceMasterKey> encryptingStream = crypto.createEncryptingStream(masterKey, in, context);
+        // 7. Copy the encrypted data into a file.
+        final File encryptedFile = new File(srcFile + ".encrypted");
+        try (FileOutputStream out = new FileOutputStream(encryptedFile)) {
+            IOUtils.copy(encryptingStream, out);
+            encryptingStream.close();
+        }
 
-        FileOutputStream out = new FileOutputStream(srcFile + ".encrypted");
-        IOUtils.copy(encryptingStream, out);
-        encryptingStream.close();
-        out.close();
+        // 8. Create the decrypting stream.
+        final AwsCryptoInputStream decryptingStream =
+                crypto.createDecryptingStream(config, new FileInputStream(encryptedFile));
```

Comment:
I was about to comment that this should use a try-with-resources too, but then remembered that only works if you're directly instantiating the object :(

In any case, you'll want a manual try/finally to make sure this is closed as well.

Reply:
I tested it out with try-with-resources and it seems to auto close even without directly instantiating the AwsCryptoInputStream

Reply:
Ah, now I recall that you just don't get the IDE warning unless you're instantiating the type directly inside the try (...).

Reply:
I've updated to use try-with-resources for the AwsCryptoInputStream and AwsCryptoOutputStream. I realize I could put both the AwsCrypto streams and the FiltOutputStreams into the same try-with-resources statement like:

``` java
try(final AwsCryptoInputStream decryptingStream = crypto.createDecryptingStream(config, new FileInputStream(encryptedFile));
           final FileOutputStream out = new FileOutputStream(decryptedFile)) {
```

, but it didn't flow as well for me in terms of delineating the steps, so I separated them. Let me know if you'd prefer them together though